### PR TITLE
Evaluate exclude regex when streaming zipped dir

### PR DESCRIPTION
**Pull request by zmcode**

### CODE REVIEW

File: pydirl/files_utils.py [hunk 1]
```diff
@@ -31,9 +31,12 @@ def get_mtime(path):
     return os.path.getmtime(path)
 
 
-def directory_to_zipstream(path):
+def directory_to_zipstream(path, exclude=None):
```
Comment (ael-code) [inline pydirl/files_utils.py]: Minor concern: for modularity I would say we shouldn't expect the exclude parameter to be a regexp already compiled.

File: pydirl/files_utils.py [hunk 2]
```diff
@@ -42,6 +45,9 @@ def directory_to_zipstream(path):
             if not (os.path.isfile(absPath) or os.path.isdir(absPath)):
                 logger.debug('Skipping unknown element: {}'.format(absPath))
                 continue
+            if exclude and exclude.match(file):
```
Comment (ael-code) [inline pydirl/files_utils.py]: We can avoid to check the file if we know that won't be included in the stream. I mean we can make as first check the regexp one and then the others.

**Comment by ael-code** [comment]
Ops I've just find out this:
```
  File "/home/travis/build/ael-code/pydirl/pydirl/files_utils.py", line 38, in directory_to_zipstream

    logger.debug("Excluding element: '{0}'".format(file))

UnboundLocalError: local variable 'file' referenced before assignment
```

**Comment by ael-code** [comment]
In any case I've implemented tests for this part of the code. You just need to merge my branch `iss16`

File: pydirl/files_utils.py (hunk 2 above)
Comment (zmcode) [inline pydirl/files_utils.py]: Makes sense.

**Comment by zmcode** [comment]
```
Traceback (most recent call last):
  File "/home/travis/build/ael-code/pydirl/test/test_zipped.py", line 31, in test_zipped
    with ZipFile(zippedRoot) as zf:
AttributeError: ZipFile instance has no attribute '__exit__'
```
`ZipFile` cannot be used in `with` statement in python 2.6 because it misses the `__exit__` method

**Comment by ael-code** [comment]
On October 29, 2013 there was the last python 2.6 release and then it was discontinued. I think that we can drop support for this outdated version of python. This will allow us to write better and more concise code.

**Comment by ael-code** [comment]
ael-code wrote:
> On October 29, 2013 there was the last python 2.6 release and then it was discontinued. I think that we can drop support for this outdated version of python. This will allow us to write better and more concise code.

I've noticed that both python-zipstream and flask are still supporting python 2.6. Thus I quickly change my mind. You can find the test code that support python 2.6 on the same branch iss16

**Comment by zmcode** [comment]
@ael-code Everything seems fine now.

**Pull request merged**